wiki/QFS-Client-Reference.md

```diff
@@ -283,3 +283,92 @@ _write-behind threshold_ by subsequent write calls or until user calls
 * _numBytes + pending >= write-behind threshold_: QFS client makes
 a copy of the source buffer only if _write-behind threshold_ is greater than
 zero. Write is performed in a blocking fashion.
+
+## Append Operations
+
+### RecordAppend
+
+RecordAppend can be used by a single writer to append to a replicated file. There is
+limited support for append to RS files.
+
+The file should be opened using the mode O_APPEND or O_WRONLY | O_APPEND.
+When RecordAppend is used to write a record, the entire record is guaranteed to be
+written to the same chunk. If the record will not fit in the current chunk, a new chunk
+is started. Note that this will generally create a sparse file since the new chunk will
+start at the next chunk boundary location.
+
+For example, using default settings, the sequence
+
+    char data[] = { '1', '2', '3' };
+    int fd = client->Open(filename, O_CREAT | O_EXCL);
+    client->Write(fd, &data[0], 1);
+    client->Close(fd);
+    
+    fd = client->Open(filename, O_WRONLY | O_APPEND);
+    client->RecordAppend(fd, &data[1], 1);
+    client->RecordAppend(fd, &data[2], 1);
+    client->Close(fd);
+
+will result in a file with 2 chunks, each replicated 3 times, and with an size of
+134217728 bytes.
+
+An attempt to open this file for read will stall until all of the chunks are considered
+stable.
+
```

Comment:
The file can be open for read anytime, however only "stable" chunks are available for read, as non stable chunks will either have write lease, or will be in transitioning to stable. One can think (and use) file opened for append as a message queue, and chunks as items in this queue. The once chunk transitions into stable state it can be read.